### PR TITLE
[MC 0.5] -  Add Account management actions

### PR DESCRIPTION
**Pull request by tommasini**
**Description**
This PR aims to add a new sheet for account management actions, such as adding an account, importing an account or adding a hardware wallet.

**Screenshots/Recordings**
Full behaviour: https://recordit.co/rxG9CBO0E4 (The button when connected to a dapp was updated to a link button after this was recorded, all the recorders on the test cases are with the last version)

Test cases:
Scenario: The account import/creation button should exist on the account selector sheet when accessed from the Wallet view
https://recordit.co/SL47lzhYSP

Scenario: The account import/creation button should exist on the account selector sheet when accessed from the Browser view while connected to a Dapp
https://recordit.co/fAEd7ua2li

Scenario: The account import/creation button should exist on the account selector sheet when accessed from the Browser view before connecting to a Dapp
https://recordit.co/fAEd7ua2li (Last 15 seconds)

Scenario: The account import/creation button should exist on the account selector sheet when accessed from the Browser view
repeated: https://recordit.co/fAEd7ua2li

Scenario: New account should be created when Add new account is tapped from the account additions sheet
Both recordings above show this scenario

Scenario: The account should be imported when the Import account is tapped from the account additions sheet
https://recordit.co/zGUJhG7oPe

Scenario: Hardware wallet should be connected when Add hardware wallet is tapped from the account additions sheet
**Needs to be tested with someone who has a hardware wallet**

**Issue**

Progresses #???

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: app/components/Views/AccountConnect/AccountConnectMultiSelector/AccountConnectMultiSelector.tsx [hunk 1]
```diff
@@ -12,16 +11,21 @@ import { useStyles } from '../../../../component-library/hooks';
...
 import generateTestId from '../../../../../wdio/utils/generateTestId';
 import { ACCOUNT_APPROVAL_SELECT_ALL_BUTTON } from '../../../../../wdio/screen-objects/testIDs/Components/AccountApprovalModal.testIds';
 
+// Internal dependencies.
+import styleSheet from './AccountConnectMultiSelector.styles';
+import {
+  AccountConnectMultiSelectorProps,
+  AccountConnectMultiSelectorScreens,
+} from './AccountConnectMultiSelector.types';
+import AddAccountActions from '../../AddAccountActions/AddAccountActions';
```
Comment (Cal-L) [inline app/components/Views/AccountConnect/AccountConnectMultiSelector/AccountConnectMultiSelector.tsx]: Redundant path

File: app/constants/navigation/Routes.ts [hunk 2]
```diff
@@ -64,6 +64,7 @@ const Routes = {
     ACCOUNT_PERMISSIONS: 'AccountPermissions',
     NETWORK_SELECTOR: 'NetworkSelector',
     ACCOUNT_ACTIONS: 'AccountActions',
+    ADD_ACCOUNT_ACTIONS: 'AddAccountActions',
```
Comment (Cal-L) [inline app/constants/navigation/Routes.ts]: No need, remove

File: app/components/Views/NetworkSelector/NetworkSelector.tsx [hunk 3]
```diff
@@ -269,7 +272,12 @@ const NetworkSelector = () => {
 
   return (
     <SheetBottom ref={sheetRef}>
-      <SheetHeader title={strings('networks.select_network')} />
+      <SheetHeader
+        title={strings('networks.select_network')}
+        onBack={
+          fromAccountPermission ? () => sheetRef.current?.hide() : undefined
```
Comment (Cal-L) [inline app/components/Views/NetworkSelector/NetworkSelector.tsx:278]: TBD - may either remove or change to X on the right instead
Thread - https://consensys.slack.com/archives/C04P4B4JA3T/p1685056685942669?thread_ts=1684852201.011759&cid=C04P4B4JA3T

File: app/components/Views/NetworkSelector/NetworkSelector.types.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export interface NetworkSelectorParams {
+  fromAccountPermission?: boolean;
```
Comment (Cal-L) [inline app/components/Views/NetworkSelector/NetworkSelector.types.ts:2]: TBD

File: app/components/Views/AddAccountActions/AddAccountActions.styles.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+// Third party dependencies.
+import { StyleSheet } from 'react-native';
+
+/**
+ * Style sheet function for AddAccountActions component.
+ *
+ * @returns StyleSheet object.
+ */
+const styleSheet = () =>
```
Comment (Cal-L) [inline app/components/Views/AddAccountActions/AddAccountActions.styles.ts]: Don't need to be function

File: app/components/Views/AddAccountActions/AddAccountActions.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import Logger from '../../../util/Logger';
+import Engine from '../../../core/Engine';
+
+// Internal dependencies
+import styleSheet from './AddAccountActions.styles';
+import { AddAccountActionsProps } from './AddAccountActions.types';
+
+const AddAccountActions = ({ onBack }: AddAccountActionsProps) => {
+  const { navigate } = useNavigation();
+  const { styles } = useStyles(styleSheet, {});
```
Comment (Cal-L) [inline app/components/Views/AddAccountActions/AddAccountActions.tsx]: No need to use useStyles

File: app/components/Views/AddAccountActions/AddAccountActions.styles.ts (hunk 5 above)
Comment (Cal-L) [reply app/components/Views/AddAccountActions/AddAccountActions.styles.ts]: We can remove this style since it's not needed

**Review by Cal-L — COMMENTED** [review]
Left some comments

File: app/components/Views/NetworkSelector/NetworkSelector.tsx (hunk 3 above)
Comment (Cal-L) [reply app/components/Views/NetworkSelector/NetworkSelector.tsx:278]: Remove back button for now

File: app/components/Views/NetworkSelector/NetworkSelector.types.ts (hunk 4 above)
Comment (Cal-L) [reply app/components/Views/NetworkSelector/NetworkSelector.types.ts:2]: Remove this as well

**Review by Cal-L — DISMISSED** [review]
LGTM

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**